finetune/base.py

```diff
@@ -245,6 +246,42 @@ def _distribute_strategy(self, visible_gpus):
         self.resolved_gpus = resolved_gpus
         return distribute_strategy
 
+
+    def load_separate(self,path):
+        """
+        Initialize a model with target and featurizer split into two estimators. This allows
+        faster loading and changing weights within a constant architecture in deployment.
+        """
+        assert self.config.build_separate_estimators,\
+        "Must have 'build separate estimators' enabled in config for post-initialization loading."
+        assert self.config.bert_adapter_size is not None,\
+        "Must have adapters turned on in config to keep loaded file size low"
+        self._load_path = path
+        orig = self.saver.load(path)
+        self.need_to_refresh = True
+        self.input_pipeline = orig.input_pipeline
+        self.get_separate_estimators(path)
+
+    def change_weights(self,path):
+        """
+        Initialize a model with target and featurizer split into two estimators. This allows
+        faster loading and changing weights within a constant neural architecture in deployment.
+        """
+        assert hasattr(self,'featurizer_est') and hasattr(self,'target_est'), "You must call \
+        load_separate to initialize the model before switching out weights"
+        self.need_to_refresh = True
+        orig = self.saver.load(path)
+        self.input_pipeline = orig.input_pipeline
+        self._load_path = path
+        del self.featurizer_est
+        del self.target_est
+        self.get_separate_estimators(path)
+        for hook in self.hooks:
```

Comment:
This implies that the only hooks will be the saver hooks. If you rename self.hooks to be saver_hooks then this seems fine. Otherwise we will have issues when people try to add other hooks